Core/include/Acts/Propagator/detail/CovarianceEngine.hpp

```diff
@@ -24,6 +24,25 @@ namespace Acts {
 /// calculations are identical for @c StraightLineStepper and @c EigenStepper.
 /// As a consequence the methods can be located in a seperate file.
 namespace detail {
+using Covariance = std::variant<BoundSymMatrix, FreeSymMatrix>;
+using Jacobian =
+    std::variant<BoundMatrix, FreeToBoundMatrix, BoundToFreeMatrix, FreeMatrix>;
+
+/// @brief Constructs a jacobian to transform from (x,y,z,t,phi,theta,q/p) to
+/// (x,y,z,t,Tx,Ty,Tz,q/p)
+///
+/// @param [in] dir Direction vector
+///
+/// @return The jacobian
+ActsMatrixD<8, 7> jacobianDirectionsToAngles(const Vector3D& dir);
```

Comment:
Maybe I misunderstand something, but I guess the name of this function should be `jacobianAnglesToDirections`...

Reply:
The naming is a bit misleading, you're right. Since it is a transformation that is multiplied from the right side to the jacobian it actually transforms the dependency as the naming suggests. I will think about renaming them.